Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm

```diff
@@ -1486,79 +1493,68 @@ static void webProcessPoolHighDynamicRangeDidChangeCallback(CFNotificationCenter
+        process.send(Messages::WebProcess::RegisterFontMap(*m_userInstalledFontURLs, *m_userInstalledFontFamilyMap,  sandboxExtensionsForUserInstalledFonts(*m_sandboxExtensionURLs, process.auditToken())), 0);
         return;
     }
 
-    RetainPtr userInstalledFontsPath = [NSHomeDirectory() stringByAppendingPathComponent:@"Library/Fonts"];
-
     HashMap<String, URL> fontURLs;
+    HashMap<String, Vector<String>> fontFamilyMap;
     Vector<URL> sandboxExtensionURLs;
-    addUserInstalledFontURLs(userInstalledFontsPath.get(), fontURLs, sandboxExtensionURLs);
-    addUserInstalledFontURLs(@"/Library/Fonts", fontURLs, sandboxExtensionURLs);
-    addUserInstalledFontURLs(@"/System/Library/AssetsV2/com_apple_MobileAsset_Font7", fontURLs, sandboxExtensionURLs);
 
-    process.send(Messages::WebProcess::RegisterFontMap(fontURLs, sandboxExtensionsForUserInstalledFonts(sandboxExtensionURLs, process.auditToken())), 0);
+    RELEASE_LOG(Process, "WebProcessPool::registerUserInstalledFonts: start registering fonts");
+    RetainPtr requestedProperties = [NSSet setWithArray:@[@"NSFontNameAttribute", @"NSFontFamilyAttribute", @"NSCTFontFileURLAttribute", @"NSCTFontUserInstalledAttribute"]];
+    RetainPtr fontProperties = XTCopyPropertiesForAllFonts(bridge_cast(requestedProperties.get()), kXTScopeAll);
+    if (!fontProperties)
+        return;
+    for (CFIndex i = 0; i < CFArrayGetCount(fontProperties.get()); ++i) {
+        RetainPtr fontDictionary = checked_cf_cast<CFDictionaryRef>(CFArrayGetValueAtIndex(fontProperties.get(), i));
+        if (!fontDictionary)
+            continue;
+        RetainPtr cfFontURL = checked_cf_cast<CFURLRef>(CFDictionaryGetValue(fontDictionary.get(), CFSTR("NSCTFontFileURLAttribute")));
+        URL fontURL(cfFontURL.get());
+        if (fontURL.string().startsWith("file:///System/Library/Fonts/"_s))
+            continue;
+        if (fontURL.string().startsWith("file:///System/Library/PrivateFrameworks/"_s))
+            continue;
+        RetainPtr fontNameAttribute = checked_cf_cast<CFStringRef>(CFDictionaryGetValue(fontDictionary.get(), CFSTR("NSFontNameAttribute")));
+        RetainPtr fontFamilyNameAttribute = checked_cf_cast<CFStringRef>(CFDictionaryGetValue(fontDictionary.get(), CFSTR("NSFontFamilyAttribute")));
+        String fontName(fontNameAttribute.get());
+        String fontFamilyName(fontFamilyNameAttribute.get());
+        auto fontNameLowerCase = fontName.convertToASCIILowercase();
+        if (fontNameLowerCase.isEmpty())
+            continue;
+        fontURLs.add(fontNameLowerCase, fontURL);
+        auto fontFamilyNameLowerCase = fontFamilyName.convertToASCIILowercase();
+        if (fontFamilyNameLowerCase.isEmpty())
+            continue;
+        auto fontNames = fontFamilyMap.find(fontFamilyNameLowerCase);
+        if (fontNames != fontFamilyMap.end())
+            fontNames->value.append(fontNameLowerCase);
+        else {
+            Vector<String> fontNames;
```

Comment:
`Vector<String> fontNames { fontNameLowerCase };` I think would be and would be more efficient then appending on the next line.

Reply:
Ah, good point, fixed in latest patch.

Thanks for reviewing!